packages/engine/lib/error/src/ext.rs

```diff
@@ -0,0 +1,111 @@
+use core::fmt;
+
+use provider::Provider;
+
+#[cfg(feature = "std")]
+use crate::Report;
+use crate::{Result, ResultExt};
+
+#[cfg(feature = "std")]
+impl<T, E> ResultExt<T> for std::result::Result<T, E>
+where
+    E: std::error::Error + Send + Sync + 'static,
+{
+    type Context = ();
+
+    #[track_caller]
+    fn wrap_err<M>(self, message: M) -> Result<T>
+    where
+        M: fmt::Display + fmt::Debug + Send + Sync + 'static,
+    {
+        match self {
+            Ok(t) => Ok(t),
+            Err(error) => Err(Report::from(error).wrap(message)),
+        }
+    }
+
+    #[track_caller]
+    fn wrap_err_lazy<M, F>(self, message: F) -> Result<T>
+    where
+        M: fmt::Display + fmt::Debug + Send + Sync + 'static,
+        F: FnOnce() -> M,
+    {
+        match self {
+            Ok(t) => Ok(t),
+            Err(error) => Err(Report::from(error).wrap(message())),
+        }
+    }
+
+    #[track_caller]
+    fn provide_context<C>(self, context: C) -> Result<T, C>
+    where
+        C: Provider + fmt::Display + fmt::Debug + Send + Sync + 'static,
```

Comment:
I would move the `Context` trait into this file to shorten this to `C: Context`, though the associated type of `ResultExt` would need to be renamed, e.g. to `ResultContext`.

Reply:
See above